application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/TaskMacroUpdateEventListener.java

```diff
@@ -0,0 +1,219 @@
+                List<DocumentReference> currentTasksIds =
+                    tasks.stream().map(Task::getReference).collect(Collectors.toList());
+                previousDocTasks.removeIf(task -> currentTasksIds.contains(task.getReference()));
+            } catch (XWikiException ignored) {
+                logger.warn(
+                    "Could not check for the possibly removed Task Macros and delete their associated Task Pages.");
+            }
+        }
+        if (tasks.size() > 0 || previousDocTasks.size() > 0) {
+            // TaskExtractor will add new IDs to the tasks if they don't have one, so we need to update the content.
+            try {
+                document.setContent(documentContent);
+            } catch (XWikiException e) {
+                logger.error("Could not update the content of the document!");
+            }
+            context.put(TASK_UPDATE_FLAG, true);
+            deleteTaskPages(document, context, previousDocTasks);
+            createOrUpdateTaskPages(document, context, tasks);
+            context.put(TASK_UPDATE_FLAG, null);
+        }
+    }
+
+    private void deleteTaskPages(XWikiDocument document, XWikiContext context, List<Task> previousDocTasks)
+    {
+        for (Task previousDocTask : previousDocTasks) {
+            try {
+                XWikiDocument taskDoc = context.getWiki().getDocument(previousDocTask.getReference(), context);
+                BaseObject taskObj = taskDoc.getXObject(TASK_CLASS_REFERENCE);
+                if (!document.getDocumentReference()
+                    .equals(resolver.resolve(taskObj.getLargeStringValue(Task.OWNER))))
+                {
+                    continue;
+                }
+                if (authorizationManager.hasAccess(Right.DELETE, context.getUserReference(),
+                    previousDocTask.getReference()))
+                {
+                    context.getWiki().deleteDocument(taskDoc, context);
+                } else if (authorizationManager.hasAccess(Right.EDIT, context.getUserReference(),
+                    previousDocTask.getOwner()))
+                {
+                    taskObj.set(Task.OWNER, "", context);
+                    context.getWiki().saveDocument(taskDoc, context);
+                } else {
+                    logger.warn(
+                        "The task macro with id [{}] was removed but the associated page could not be deleted or "
+                            + "modified.",
+                        previousDocTask.getReference());
+                }
+            } catch (XWikiException e) {
+                logger.error("Failed to remove the Task Document with id [{}]", previousDocTask.getReference());
+            }
+        }
+    }
+
+    private void createOrUpdateTaskPages(XWikiDocument document, XWikiContext context, List<Task> tasks)
+    {
+        for (Task task : tasks) {
+            DocumentReference macroReference = task.getReference();
```

Comment:
```suggestion
            DocumentReference taksReference = task.getReference();
```

``macroReference`` is confusing.